contracts/multisig-prover/src/execute.rs

```diff
@@ -129,31 +132,50 @@ fn get_verifiers_info(deps: &DepsMut, config: &Config) -> Result<VerifiersInfo,
             msg: to_json_binary(&active_verifiers_query)?,
         }))?;
 
-    let participants = verifiers
-        .clone()
+    let min_num_verifiers = deps
+        .querier
+        .query::<Service>(&QueryRequest::Wasm(WasmQuery::Smart {
+            contract_addr: config.service_registry.to_string(),
+            msg: to_json_binary(&service_registry::msg::QueryMsg::GetService {
+                service_name: config.service_name.clone(),
+            })?,
+        }))?
+        .min_num_verifiers;
+
+    let participants_with_pubkeys = verifiers
         .into_iter()
-        .map(WeightedVerifier::into)
-        .collect::<Vec<snapshot::Participant>>();
-
-    let snapshot =
-        snapshot::Snapshot::new(config.signing_threshold, participants.clone().try_into()?);
-
-    let mut pub_keys = vec![];
-    for participant in &participants {
-        let pub_key_query = multisig::msg::QueryMsg::GetPublicKey {
-            verifier_address: participant.address.to_string(),
-            key_type: config.key_type,
-        };
-        let pub_key: PublicKey = deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
-            contract_addr: config.multisig.to_string(),
-            msg: to_json_binary(&pub_key_query)?,
-        }))?;
-        pub_keys.push(pub_key);
+        .filter_map(|verifier| {
+            let pub_key_query = multisig::msg::QueryMsg::GetPublicKey {
+                verifier_address: verifier.verifier_info.address.to_string(),
+                key_type: config.key_type,
+            };
+
+            match deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
+                contract_addr: config.multisig.to_string(),
+                msg: to_json_binary(&pub_key_query).ok()?,
+            })) {
+                Ok(pub_key) => Some((Participant::from(verifier), pub_key)),
+                Err(_) => None,
+            }
+        })
+        .collect::<Vec<_>>();
+
+    if participants_with_pubkeys.len() < min_num_verifiers as usize {
+        return Err(ContractError::NotEnoughVerifiers);
     }
 
+    let snapshot = Snapshot::new(
+        config.signing_threshold,
+        participants_with_pubkeys
+            .iter()
+            .map(|(participant, _)| participant.clone())
+            .collect::<Vec<_>>()
+            .try_into()?,
+    );
+
     Ok(VerifiersInfo {
         snapshot,
-        pubkeys_by_participant: participants.into_iter().zip(pub_keys).collect(),
+        pubkeys_by_participant: participants_with_pubkeys,
     })
```

Comment:
I think we should get rid of this type, I don't see why we can't just directly create the `VerifierSet`. We have too many types that are basically the same thing, and it's quite confusing. I think this is the only place it's used.

Reply:
let me try to remove